### PR TITLE
Updating to support BuildMaster 6.2.33.

### PR DESCRIPTION
**Pull request by splatteredbits**
* Pipelines_* native API replaced with Rafts_* API (i.e. pipelines are now stored in rafts as raft items).
* Packages renamed to Builds.

### CODE REVIEW

File: BuildMasterAutomation/Functions/ConvertTo-BMNativeApiByteValue.ps1 [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    .DESCRIPTION
+    Some of the parameters of the BuildMaster native API are typed as `byte[]` object. This function converts strings
+    into a value that can be passed as one of these parameters. Pipe the string you want to convert (or pass it to the
+    `InputObject` parameter). The function will return a value that you can pass to the BuildMaster API.
+
+    If you pipe multiple strings to `ConvertTo-BMNativeApiByteValue`, the strings will be concatenated together before
+    conversion.
+
+    .EXAMPLE
+    ConvertFrom-BMNativeApiByteValue 'hello example'
```
Comment (decoyjoe) [inline BuildMasterAutomation/Functions/ConvertTo-BMNativeApiByteValue.ps1]: Typo: `ConvertFrom` should be `ConvertTo`

File: BuildMasterAutomation/Functions/Get-BMRaftItem.ps1 [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    are returned.
+    #>
+    [CmdletBinding()]
+    param(
+        # A session object to the BuildMaster instance to use. Use the `New-BMSession` function to create a session.
+        [Parameter(Mandatory)]
+        [Object] $Session,
+
+        [Parameter(Mandatory)]
+        # The raft id, name, or object whose items to return.
```
Comment (decoyjoe) [inline BuildMasterAutomation/Functions/Get-BMRaftItem.ps1:47]: Nit: Parameter doc should be above the parameter attribute.

File: BuildMasterAutomation/Functions/Invoke-BMRestMethod.ps1 [hunk 3]
```diff
@@ -6,65 +6,69 @@ function Invoke-BMRestMethod
...
         # The HTTP/web method to use. The default is `GET`.
-        $Method = [Microsoft.PowerShell.Commands.WebRequestMethod]::Get,
+        [Microsoft.PowerShell.Commands.WebRequestMethod] $Method =
+            [Microsoft.PowerShell.Commands.WebRequestMethod]::Get,
 
-        [Parameter(Mandatory=$true,ParameterSetName='BodyFromHashtable')]
-        [hashtable]
-        # That parameters to pass to the method. These are converted to JSON and sent to the API in the body of the request.
-        $Parameter,
+        # That parameters to pass to the method. These are converted to JSON and sent to the API in the body of the
```
Comment (decoyjoe) [inline BuildMasterAutomation/Functions/Invoke-BMRestMethod.ps1]: Typo: `That` should be `The`

File: BuildMasterAutomation/Functions/Remove-BMApplication.ps1 [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    )
+
+    process
+    {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+        if (-not $Application)
+        {
+            return
```
Comment (decoyjoe) [inline BuildMasterAutomation/Functions/Remove-BMApplication.ps1]: Should we write a message here?

File: BuildMasterAutomation/Functions/Set-BMRaftItem.ps1 [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    .EXAMPLE
+    Set-BMRaftItem -Session $session -Raft $raft -TypeCode Pipeline -Name 'Pipeline'
+
+    Demonstrates how to use Set-BMRaftItem. In this example, an empty, global pipeline will be created in the raft
+    represented by the `$raft` object.
+
+    .EXAMPLE
+    Set-BMRaftItem -Session $session -Raft $raft -TypeCode Module -Name 'Module' -Application $app -Content $otterScript
+
+    Demonstrates how to use Set-BMRaftItem. In this example, an module will be created/updated for the application
```
Comment (decoyjoe) [inline BuildMasterAutomation/Functions/Set-BMRaftItem.ps1]: Typo: `an module` should be `a module`

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,125 @@
...
+* Rename usages of the `New-BMPipeline` function to `Set-BMPipeline`.
+* `Set-BMPipeline` (née `New-BMPipeline`) now creates *or* updates a pipeline. Inspect usages to see the impact of this
+new behavior.
+* Update usages of the `Set-BMPipeline` (née `New-BMPipeline`) function's `Stage` parameter to pass stage objects
+instead of strings of XML. Use the new `New-BMPipelineStageObject` and `New-BMPipelineStageTargetObject` functions to
+create the objects you should pass.
+* Update usages of the `Set-BMPipeline` (née `New-BMPipeline`) function that use the return value. `Set-BMPipeline` no
+longer returns the pipeline object by default. Use the new `PassThru` switch and the pipeline object will be returned.
+* Add `-ErrorAction Ignore` to usages of `Remove-BMServer` and `Remove-BMServerRole` if you don't care if the item doesn't exist. `Remove-BMServer` and `Remove-BMServerRole` now write errors when the item to remove doesn't exist.
+* Update usages of `Set-BMRelease` to pass a pipeline name or pipeline object to the new `Pipeline` paramter and remove
```
Comment (decoyjoe) [inline CHANGELOG.md:33]: Typo: `paramter`

**Review by decoyjoe — DISMISSED** [review]

File: BuildMasterAutomation/Functions/Remove-BMApplication.ps1 (hunk 4 above)
Comment (splatteredbits) [reply BuildMasterAutomation/Functions/Remove-BMApplication.ps1]: Not sure why this check is here. The application is a mandatory parameter. Removed.

**Pull request merged**